### PR TITLE
refactor(uibackend): create new uibackend module

### PR DESCRIPTION
**Pull request by paralta**
## Description

https://github.com/openclarity/vmclarity/blob/main/rfc/multiple-go-modules.md

* Create new `uibackend` module
* Update licensei to ignore gen.go files and not directories

## Type of Change

[ ] Bug Fix  
[ ] New Feature  
[ ] Breaking Change  
[x] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

**Comment by chrisgacsal** [comment]
> Two notes:
> 
> `uibackend` package is used to generate API models only, but is defined on a parent level (above server and client). Is it possible to have `uibackend/api/models`, `uibackend/client`, `uibackend/server` instead of the current mode? It makes it a bit cleaner to separate the package responsibility

This sounds like a good idea. I would probably go even further and rename `models` to `types`. e.g. `uibackend/types`
Same should be followed for `api` module as well.

>     * `uibackend/server/pkg` can be moved to `uibackend/server` to keep it clean similar to `uibackend/client`

Disagree. I would keep  `uibackend/server/pkg` as there is a `uibackend/server/cmd` package as well. the `server` is an actual applications while the `client` is "just" a library.

File: uibackend/server/go.mod [hunk 1]
```diff
@@ -0,0 +1,75 @@
+module github.com/openclarity/vmclarity/uibackend/server
+
+go 1.21.4
+
+replace (
```
Comment (chrisgacsal) [inline uibackend/server/go.mod:5]: `replace` directives should be added after the `require` directives, at the bottom of the file in this case.

**Comment by paralta** [comment]
thanks for reviewing, @ramizpolic and @chrisgacsal ! I will make the changes mentioned and split `uibackend` into 3 modules: `types` instead of `models`, `client` and `server` 👍

**Comment by chrisgacsal** [comment]
Please hold on with merging this until the following PR is merged: https://github.com/openclarity/vmclarity/pull/864

File: uibackend/server/pkg/common.go [hunk 2]
```diff
@@ -13,13 +13,13 @@
...
 
-package rest
+package server
 
 import (
 	"github.com/labstack/echo/v4"
 	log "github.com/sirupsen/logrus"
 
-	"github.com/openclarity/vmclarity/pkg/uibackend/api/models"
+	"github.com/openclarity/vmclarity/api/models"
```
[the remark this answers is not in the document]
Comment (paralta) [reply uibackend/server/pkg/common.go:22]: nice catch! definitely an error

**Review by chrisgacsal — APPROVED** [review]
LGTM! :shipit:

**Review by chrisgacsal — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**